google-cloud-bigquery/lib/google/cloud/bigquery/schema.rb

```diff
@@ -123,6 +184,84 @@ def empty?
           fields.empty?
         end
 
+        ##
+        # Load the schema from a JSON file.
+        #
+        # The JSON schema file is the same as for the [`bq`
+        # CLI](https://cloud.google.com/bigquery/docs/schemas#specifying_a_json_schema_file)
+        # consisting of an array of JSON objects containing the following:
+        # - `name`: The column's [data
+        #   type](https://cloud.google.com/bigquery/docs/schemas#standard_sql_data_types)
+        # - `type`: The column [name](https://cloud.google.com/bigquery/docs/schemas#column_names)
+        # - `description`: (Optional) The column's [description](https://cloud.google.com/bigquery/docs/schemas#column_descriptions)
+        # - `mode`: (Optional) The column's [mode](https://cloud.google.com/bigquery/docs/schemas#modes)
+        #   (if unspecified, mode defaults to `NULLABLE`)
+        # - `fields`: If `type` is `RECORD`, an array of objects defining child
+        #   fields with these properties
+        #
+        # @param [IO, String, Array<Hash>] source An `IO` containing the JSON
+        #   schema, a `String` containing the JSON schema, or an `Array` of
+        #   `Hash`es containing the schema details.
+        #
+        # @return [Schema] The schema so that commands are chainable.
+        #
+        # @example
+        #   require "google/cloud/bigquery"
+        #
+        #   bigquery = Google::Cloud::Bigquery.new
+        #   dataset = bigquery.dataset "my_dataset"
+        #   table = dataset.table "my_table" do |table|
+        #     table.schema.load File.read("path/to/schema.json")
+        #   end
+        #
+        def load source
+          if source.respond_to?(:rewind) && source.respond_to?(:read)
+            source.rewind
+            schema_json = String source.read
+          elsif source.is_a? Array
+            schema_json = JSON.dump source
+          else
+            schema_json = String source
+          end
+
+          schema_json = %({"fields":#{schema_json}})
+          raise ArgumentError, "Invalid JSON" unless schema_json
```

Comment:
This `raise` can be removed. It will never be `nil` because of the previous line.